src/schema.ts

```diff
@@ -9,13 +10,18 @@ export const urlSchema = z
   .string()
   .url()
   .startsWith(urlStart, `URL must start with [${urlStart}].`)
-  //.co will cause fetch to time out
-  .endsWith(urlEnd, `URL must end with [${urlEnd}].`)
   .min(
     urlStart.length + urlEnd.length + 1,
     'Domain name must be at least 1 character long.'
   )
   .refine(
     (url) => !url.endsWith('-admin.okta.com'),
     'Organisation URL should not be the admin URL. Please remove "-admin" and try again.'
+  )
+  .refine(
+    //.co will cause fetch to time out
+    (url) => {
+      return readonlyURL(url)?.hostname.endsWith(urlEnd);
```

Comment:
**Question:** shouldn't line 18 also change? 

By the time you get to this point, it would be ideal if you have parsed the string into a stronger type. You need a `?` on this line because that parsing hasn't happened?

Reply:
The ? is because it returns a ReadonlyURL|undefined

Reply:
Yep. I understand that. If you look at https://github.com/colinhacks/zod/blob/master/src/types.ts#L768 you will see that zod is not returning a stricter type here. You get a string back.

That means that the rest of your code (i.e. this line here) can't rely on what zod has already done.

Take 5 minutes to try using https://zod.dev/?id=validating-during-transform as an approach to get a `ReadonlyURL` directly from zod.

Reply:
I just tried using transform. I'm running into the issue where
* if i parse first, i can't check for the double / in https://
* if I do the startswith on the string then parse, if https:// was give, zod would skip the transform, and refinement which relied accessing the parsedUrl will throw an error.